Add Gammu based local SMS notifications

ocalvo opened the pull request:
## Breaking Change:
No

## Description:
This creates a new notification component to send SMS via a GSM modem without needing an active internet connection.
More details [here](https://github.com/ocalvo/ha-sms/edit/master/README.md)

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request with documentation for [home-assistant.io](https://github.com/home-assistant/home-assistant.io) (if applicable):** home-assistant/home-assistant.io#11826

[Documentation PR](https://github.com/home-assistant/home-assistant.io/pull/11826)

## Example entry for `configuration.yaml` (if applicable):

```yaml
sms:
  device: /dev/ttyUSB2

notify:
  - name: sms_personX
    platform: sms
    recipient: "PersonX_PhoneNumber"
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
```
ERROR:   py36: commands failed
ERROR:   py37: commands failed
ERROR:   py38: commands failed
  lint: commands succeeded
  pylint: commands succeeded
  typing: commands succeeded
ERROR:   cov: commands failed
```
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] [_The manifest file_][manifest-docs] has all fields filled out correctly. Update and include derived files by running `python3 -m script.hassfest`.
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `python3 -m script.gen_requirements_all`.
  - [x] Untested files have been added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.
  - [x] Manually tested with a HUAWEI GSM dongle and an active SMS service.

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

danielperna84 committed:
Update pyhomematic to 0.1.63 (#30594)

frenck committed:
Fix discovery for oauth2 flow implementations (#30700)

* Fix discovery for oauth2 flow implementations

* Fix user step

* Add tests

bendavid committed:
update aiopylgtv to 0.2.5 (#30702)

pnbruckner committed:
Revert "Forget auth token when going offline so we can reconnect (#26630)" (#30705)

This reverts commit 2d6d6ba90e5e2f179a07c4ecd7de1744e86a8025.

balloob committed:
Fix Ring wifi sensors (#30735)

* Fix Ring wifi sensors

* Update before adding

dmulcahey committed:
Bump ZHA quirks to 0.0.31 (#30740)

* Bump ZHA quirks version

* update requirements

bendavid committed:
update aiopylgtv to 0.2.6 (#30739)

balloob committed:
Set default locale for cloud Alexa config (#30749)

pvizeli committed:
Refactor HomeMatic / Fix issue with 0.104/dev (#30752)

* Refactor HomeMatic / Fix issue with 0.104/dev

* Fix lock

alandtse committed:
Bump teslajsonpy to 0.2.3 (#30750)

frenck committed:
Fix HomeKit behavior with lights supporting color and temperature (#30756)

frenck committed:
Fix HomeKit behavior with lights supporting color and temperature (#30756)

basnijholt committed:
bump aiokef to 0.2.5 which uses locks (#30753)

basnijholt committed:
bump aiokef to 0.2.5 which uses locks (#30753)

akinomeroglu committed:
Fix small typo in alarmdotcom component (#30758)

springstan committed:
Fix supported_features in MQTT fan (#28680)

* Added custom validator function for speed list

* Replace CONF_SPEED_STATE_TOPIC with CONF_SPEED_COMMAND_TOPIC to determine SUPPORT_SET_SPEED

* Revert "Added custom validator function for speed list"

This reverts commit f000396fa642c64bde40513ea70d9915dbd71ead.

* Replace CONF_OSCILLATION_STATE_TOPIC with CONF_OSCILLATION_COMMAND_TOPIC to determine SUPPORT_OSCILLATE

garbled1 committed:
Add support for the voltage sensor on the greeneye GEM (#30484)

* Add support for the voltage sensor on the greeneye GEM

* Changed per suggestion @springstan

balloob committed:
Update Ring to 0.6.0 (#30748)

* Update Ring to 0.6.0

* Update sensor tests

* update -> async_update

* Delete temp files

* Address comments

* Final tweaks

* Remove stale print

balloob committed:
Update Ring to 0.6.0 (#30748)

* Update Ring to 0.6.0

* Update sensor tests

* update -> async_update

* Delete temp files

* Address comments

* Final tweaks

* Remove stale print

balloob committed:
Bumped version to 0.104.0b5

balloob committed:
Add Safe Mode (#30723)

* Store last working HTTP settings

* Add safe mode

* Fix tests

* Add cloud to safe mode

* Update logging text

* Fix camera tests leaving files behind

* Make emulated_hue tests not leave files behind

* Make logbook tests not leave files behind

* Make tts tests not leave files behind

* Make image_processing tests not leave files behind

* Make manual_mqtt tests not leave files behind

pvizeli committed:
Revert #29701 (#30766)

* Revert #29701

* fix format

* Fix lint

pvizeli committed:
Revert #29701 (#30766)

* Revert #29701

* fix format

* Fix lint

pvizeli committed:
Hass.io allow to reset password with CLI (#30755)

* Hass.io allow to reset passwort with CLI

* Add improvments

* fix comments

* fix lint

* Fix tests

* more tests

* Address comments

* sort imports

* fix test python37

balloob committed:
Whitelist Frenck for release

BKPepe committed:
Upgrade youtube_dl to version 2020.01.15 (#30767)

Adminiuga committed:
Use storage based collections for Timer platform (#30765)

* Use config dict for timer entity.

* Manage timer entities using collection helpers.

* Add tests.

* Make Timer duration JSON serializable.

homeassistant committed:
[ci skip] Translation update

bramkragten committed:
Search: Add search to default config and don't resolve area (#30762)

* Add search to default config and don't resolve area

* Move to frontend

* Minor update

Co-authored-by: Jason Hu <awarecan@users.noreply.github.com>

Adminiuga committed:
Allow input_* and timer component setup without config (#30772)

* Allow input_boolean setup without config.

* Allow input_number setup without config.

* Allow input_select setup without config.

* Allow input_text setup without config.

* Allow timer setup without config.

raman325 committed:
Add Config Flow support, Device Registry support, available property to vizio component (#30653)

* add config flow support, device registry support, available property

* raise PlatformNotReady if HA cant connect to device

* remove test logging statement and fix integration title

* store import and last user input values so user can see errors next to value that caused error

* add PARALLEL_UPDATES

* add missing type hints

* add missing type hints to tests

* fix options config flow title

* changes based on review

* better key name for message when cant connect

* fix missed update to key name

* fix comments

* remove logger from test which was used to debug and update test function names and docstrings to be more accurate

* add __init__.py to vizio tests module

* readded options flow and updated main component to handle options updates, set unique ID to serial, fixes based on review

* pop hass.data in media_player unload instead of in __init__ since it is set in media_player

* update requirements_all and requirements_test_all

* make unique_id key name a constant

* remove additional line breaks after docstrings

* unload entries during test_user_flow and test_import_flow tests to hopefully reduce teardown time

* try to speed up tests

* remove unnecessary code, use event bus to track options updates, move patches to pytest fixtures and fix patch scoping

* fix comment

* remove translations from commit

* suppress API error logging when checking for device availability as it can spam logs

* update requirements_all and requirements_test_all

* dont pass hass to entity since it is passed to entity anyway, remove entity unload from tests, other misc changes from review

* fix clearing listeners

* use config_entry unique ID for unique ID and use config_entry entry ID as update signal

* update config flow based on suggested changes

* update volume step on config import if it doesn't match config_entry volume step

* update config_entry data and options with new volume step value

* copy entry.data and entry.options before updating when updating config_entry

* fix test_import_entity_already_configured

jdeluyck committed:
Update emulated_roku to 0.1.9 (#30791)

* Update emulated_roku to 0.1.9

* Update requirements_all

mzdrale committed:
Update pyubee to 0.8 (#30785)

Jc2k committed:
Restore unit_of_measurement from entity registry (#30780)

* Restore unit_of_measurement from entity registry

* Lint fix

Jc2k committed:
Restore unit_of_measurement from entity registry (#30780)

* Restore unit_of_measurement from entity registry

* Lint fix

balloob committed:
Refactor Ring data handling (#30777)

* Refactor Ring data handling

* Add async_ to methods

balloob committed:
Refactor Ring data handling (#30777)

* Refactor Ring data handling

* Add async_ to methods

ochlocracy committed:
Add support for vacuums to Alexa. (#30764)

bendavid committed:
update to aiopylgtv 0.2.7 (#30797)

frenck committed:
Bumped version to 0.104.0

frenck committed:
Mark hide_entity deprecated in automation integration (#30799)

springstan committed:
Bump librouteros to 3.0.0 (#30800)

MatthewFlamm committed:
Update pynws to v0.10.1 (#30662)

* update to pynws 0.10.1

* remove unneeded raw json files

* move test helper module to const

frenck committed:
0.104.0 (#30803)

0.104.0

frenck committed:
Merge branch 'master' into dev

homeassistant committed:
[ci skip] Translation update

wolph committed:
Fixed pyyaml requirement from 5.2.0 to 5.2 (#30808)

balloob committed:
Handle no host info in ignored config entries (#30822)

springstan committed:
Fix setup error of Mikrotik (#30810)

pvizeli committed:
Fix mpd time issue (#30825)

* Fix mpd time issue

* Update homeassistant/components/mpd/media_player.py

Co-Authored-By: Franck Nijhof <git@frenck.dev>

bendavid committed:
Fix play_media in webostv (#30828)

Adminiuga committed:
Use collection helpers for input_datetime component (#30815)

* Refactor input_datetime.
Keep the state as datetime, but format accordingly to has_time and
has_date.

* Use config dict for input_datetime.
* Add tests.
* Lint
Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

balloob committed:
Handle location API being exhausted (#30798)

frenck committed:
Deprecate states UI options in group integration (#30831)

frenck committed:
Deprecate weblink integration (#30834)

Quentame committed:
Fix iCloud when no family members (issue #30829) (#30836)

frenck committed:
Deprecate hide_if_away from device_tracker (#30833)

frenck committed:
Deprecate history_graph integration (#30835)

balloob committed:
Reinstate and deprecate filename option for hue config (#30846)

bachya committed:
Fix sensor type creation with multiple Ambient weather stations (#30850)

peroyvind committed:
Fix for issue #29822 (#30849)

dshokouhi committed:
Remove unused import (#30858)

homeassistant committed:
[ci skip] Translation update

raman325 committed:
Bump pyvizio version to 0.1.1 (#30867)

* bump pyvizio version to resolve setup issue

* update requirements*.txt

cgtobi committed:
Update pyatmo to 3.2.2 and add available attribute (#30882)

* Update pyatmo to 3.2.1

* Update pyatmo to 3.2.2

* Remove accidentally added requirements

* Add availability property

Adminiuga committed:
Extract collection entity registry cleaner into a helper (#30844)

* Extract entity_registry_keeper into separate helper
* Refactor input_*, timer and person to use new helper.
* Make Mypy happy.
* Better name.

Commit:
Fix missing switch groups of HomematicIP Cloud (#30903)

dmulcahey committed:
add multistate back (#30889)

bachya committed:
Remove option to configure SimpliSafe scan interval (#30909)

* Enforce minimum scan interval for SimpliSafe

* Remove scan interval all together

* Fix tests

peternijssen committed:
remove PostNL component as it is no longer working (#30902)

ludeeus committed:
Adds icon endpoint to NO_AUTH (#30910)

* Adds icon endpoint to NO_AUTH

* Add test for no_auth icon get

* Blackout

balloob committed:
camera endpoint likes to timeout, catch it. (#30919)

bramkragten committed:
Updated frontend to 20200108.2 (#30921)

ochlocracy committed:
Fix Alexa semantics for covers with tilt support. (#30911)

* Fix Alexa semantics for covers with tilt support.

* Clarify wording.

* Korrect grammar.

Kane610 committed:
Fix issue with group unique id when normalising bridge id (#30904)

Kane610 committed:
Fix service device refresh calling state update (#30920)

balloob committed:
Fix hue accepting filename (#30924)

homeassistant committed:
[ci skip] Translation update

Adminiuga committed:
Add battery voltage state attribute for ZHA devices (#30901)

* Add battery voltage state attribute for ZHA devices.

* Pylint.

Adminiuga committed:
Refactor rounding for ZHA electrical measurement sensor (#30923)

springstan committed:
Update outdated documentation links in json files (#30916)

raman325 committed:
Use default media player device classes for vizio component (#30802)

* use media player defined device classes instead of custom ones, add options flow test, add timeout options parameter

* make options update error more generic

* fix config flow options update logic

* simplify logic for options update during import

* use platform list for load and unload

* update private config flow function name and description

* fix grammar in strings.json

* update mock config variable names to be more accurate

* remove timeout conf option, create device_class property

* update requirements

* update .coveragerc to indicate that config_flow has tests

* fix source of device_class property and move constants to const.py

* fix grammar in error message

* remove redundant device check in async_setup_entry since device connection is checked during config flow

* revert change to async_setup_entry, raise ConfigEntryNotReady if device can't be connected to

* update error text

* add more context to error text

zxdavb committed:
Add services to geniushub (#30918)

adds:
 - `set_zone_override`, and
 - `set_zone_mode`

zxdavb committed:
Add services to evohome (#29816)

springstan committed:
Revert "Pulseaudio: Changed default port from 4712 to 4713 (#28857)" (#30939)

This reverts commit e915dd0d95d6bc407e4659bfadcef1468c8434f4.

Adminiuga committed:
Don't use unit_of_measurement in state attributes. (#30941)

Commit:
Add play media support and Spotify control to Openhome (#28698)

ajmarks committed:
Minor enhancements to jewish_calendar (#30632)

* Minor enhancement to jewish_calendar:
 - Expose more halachic times from the underlying hdate module
 - Correct and standardize some transliterations

* Undo breking name change

* Add icon for talit time

Co-authored-by: Andrew Marks <52414333+amarks-coatue@users.noreply.github.com>

dmulcahey committed:
Add logical Zigbee device type to ZHA device info (#30954)

* add device type to device info
* capitalize
* use zigpy logical device type

homeassistant committed:
[ci skip] Translation update

Quentame committed:
Fix can't add multiple iCloud accounts (remove account name) (#30898)

* Fix can't add multiple iCloud accounts (remove account name)

* Update tests with flow.async_init()

jdeluyck committed:
Update emulated_roku to 0.2.0 (#30974)

balloob committed:
Improve Wemo config entry support, add device info (#30963)

* Improve config entry support, add device info

* async_dispatch_connect

* Fix I/O in event loop

* Do not raise PlatformNotReady inside dispatcher

* Make main discovery process async

* Do discovery as part of set up.

* Greatly simplify set up

* Add parallel updates to fan&switch

* mini cleanup

* Address comments

balloob committed:
Improve Wemo config entry support, add device info (#30963)

* Improve config entry support, add device info

* async_dispatch_connect

* Fix I/O in event loop

* Do not raise PlatformNotReady inside dispatcher

* Make main discovery process async

* Do discovery as part of set up.

* Greatly simplify set up

* Add parallel updates to fan&switch

* mini cleanup

* Address comments

balloob committed:
Improve Wemo config entry support, add device info (#30963)

* Improve config entry support, add device info

* async_dispatch_connect

* Fix I/O in event loop

* Do not raise PlatformNotReady inside dispatcher

* Make main discovery process async

* Do discovery as part of set up.

* Greatly simplify set up

* Add parallel updates to fan&switch

* mini cleanup

* Address comments

steve-gombos committed:
Ring camera fix (#30975)

* Fix ring camera entities

* Reverted test refresh interval

* Fix black errors

balloob committed:
Catch all Ring timeout errors (#30960)

* Catch more Ring errors

* Fix comment & Disable wifi entities by default

zxdavb committed:
initial commit (#30968)

homeassistant committed:
[ci skip] Translation update

balloob committed:
Add foundation for integration services (#30813)

* Add foundation for integration services

* Fix tests

* Remove async_get_platform

* Migrate Sonos partially to EntityPlatform.async_register_entity_service

* Tweaks

* Move other Sonos services to media player domain

* Move other Sonos services to media player domain

* Address comments

* Remove lock

* Fix typos

* Use make_entity_service_schema

* Add area extraction to async_extract_entities

Co-authored-by: Anders Melchiorsen <amelchio@nogoto.net>

raman325 committed:
Fix options update during import config flow step for vizio component (Bugfix for #30653) (#30977)

* fix options update logic during import

* add missing tests

* fix abort reasons and strings, add missing test

* combine steps when testing esn already exists

* readd removed test

* no mock_coro_func needed

* add block_until_done and assert entry options

zhumuht committed:
print component import error to logfile (#30346)

afaucogney committed:
Add Derivative component (#26456)

* create derivation component

based on integration component
remove left and right

* Update test

(was'n saved)

* add some functionnal point test

* Change derivation to derivative

* Continue migration from derivation to derivative

* Add codeowners info

* fix tests typo and values

* Improve code from reviewer

add test case
fix test values
still a prefix issue that should not

* create derivation component

based on integration component
remove left and right

* Update test

(was'n saved)

* add some functionnal point test

* Change derivation to derivative

* Continue migration from derivation to derivative

* Add codeowners info

* fix tests typo and values

* Improve code from reviewer

add test case
fix test values
still a prefix issue that should not

* Update homeassistant/components/derivative/sensor.py

Fix test issue with unit of measurement

Co-Authored-By: Santobert <tobhaase@gmail.com>

* Fix review

Move ValueError to SyntaxError

* precise state test

* un comment original tests and remove error tests

* Fix isort issue

* Fix review

- update doc link
- migrate to general const import

* Rollback import conf_unit, just defined localy

Co-authored-by: Santobert <tobhaase@gmail.com>

JeffLIrion committed:
Fix capability_attributes when supported_features is None (#30993)

* Fix capability_attributes when supported_features is None (water_heater)

* Fix capability_attributes when supported_features is None (media_player)

bachya committed:
Add state reproduction to remotes (#30990)

rdehuyss committed:
Add attributes departure_minutes and delay_minutes to the nmbs sensor (#30958)

* Improve sensor for automations

I've updated the sensor so that departure time and delay can be used in automatons.

Before, the departure time and delay time were only available wrapped in strings which makes it difficult to use them in automations. Using the extra attributes, one can easily use them in automations.

* Update homeassistant/components/nmbs/sensor.py

Co-Authored-By: springstan <46536646+springstan@users.noreply.github.com>

* Updates based on review

Changed min to minutes as requested

Co-authored-by: springstan <46536646+springstan@users.noreply.github.com>

bendavid committed:
Minor fixes for webostv (#30998)

* restore emulation of play pause toggle

* add protection to notify setup

* add state check for power off to avoid switching tv back on

basnijholt committed:
Add kef supports_on option (#30937)

* add supports_on_off option

* modify support_kef inplace

* all speakers support turning off remotely

zxdavb committed:
Small tweaks for evohome  (#31007)

frenck committed:
Adds missing strings to Almond (#31013)

frenck committed:
Adds missing strings to Withings (#31012)

frenck committed:
Fix deCONZ update entry from Hassio discovery (#31015)

* Fix deCONZ update entry from Hassio discovery

* Empty commit to re-trigger build

scop committed:
Enable some more bandit checks (#30857)

* Enable B108 (hardcoded tmp dir), address findings

* Enable B602 (subprocess popen with shell), address findings

* Enable B604 (start process with shell), address findings

* Enable B306 (mktemp), B307 (eval), and B325 (tempnam), no issues to address

Quentame committed:
Use config_entry.unique_id in iCloud (#30984)

* Use unique_id in iCloud

* Remove missed self._configuration_exists()

* Avoid breaking change

* Almost fix tests

* Add missing test

* Fix tests

doudz committed:
Bump zigpy-zigate to 0.5.1 (#31004)

* Bump zigpy-zigate to 0.5.1
Improve startup and channel setting

bachya committed:
Allow OpenUV entities to be unavailable (#31018)

* Allow OpenUV entities to be unavailable

* Empty commit to re-trigger build

zxdavb committed:
Handle ghost zones gracefully  (#31008)

ocalvo committed:
Merge pull request #1 from home-assistant/dev

Merge

ocalvo committed:
Add SMS integration

ocalvo committed:
Remove comment

homeassistant committed:
[ci skip] Translation update

ocalvo committed:
Update coverage file

ocalvo commented:
The component gammu-dev needs to be added to hassio:
```
docker exec homeassistant apk add gammu-dev
```
@pvizeli Can you help with that?

ocalvo committed:
Update doc url

BKPepe committed:
Upgrade importlib-metadata to version 1.4.0 (#31027)

BKPepe committed:
Upgrade pyyaml to 5.3 (#31026)

frenck commented:
@ocalvo Why would it need a dev lib the distribution?

Please note, we pre-build packages and provide wheels to our users.

dmulcahey committed:
Allow ZHA device creation for the Zigbee coordinator (#31032)

* allow zha device creation for coordinator

* don't let coordinator get removed

* fix truthy issue in logical device type

frenck committed:
Remove all empty *_setup_platform() from integrations (#31025)

* Remove all empty *_setup_platform() from integrations

* Fix tests for smartthings

* Fix tests for heos

ocalvo commented:
The dependency python-gammu requires it.

pvizeli commented:
Like I said, if this is merged to code base, I will care about the wheels

jnimmo committed:
Bump pyintesishome to v1.6 (#31044)

abmantis committed:
Update pyipma to 2.0 (#30746)

* update ipma component for pyipma 2.0

* fix wind speed; refactor forecast

* update requirements*.txt

* fix tests; update CODEOWNERS; update pyipma to 2.0.1

* minor changes as suggested in PR

* make lint happy

* fix mocking coroutines

* restore old unique id

* fix station lat/lon; update pyipma version

zhumuht committed:
Add xiaomi miio sensors cgllc.airmonitor.s1 & cgllc.airmonitor.b1 (#30345)

* support xiaomi miio sensor: cgllc.airmonitor.s1 & cgllc.airmonitor.b1

* support xiaomi miio sensor: cgllc.airmonitor.s1 & cgllc.airmonitor.b1

* rollback sensor,  modify air_quality

* only set air_quality props

* remove set state from "zhimi.airmonitor.v1"

* unit_of_measurement return None when the pm25 isn't reported

* import libs by isort

ocalvo committed:
Update doc comments

frenck committed:
Update issue and PR templates (#31056)

* Update issue and PR templates

* Empty commit to re-trigger build

springstan commented:
@ocalvo please run `python3 -m script.hassfest` inside of your dev environment to fix the validate error :)

ocalvo commented:
> @ocalvo please run `python3 -m script.hassfest` inside of your dev environment to fix the validate error :)

Script fails with:
```
(venv) oscar@Oscar-Surface:/mnt/c/dd/Repos/home-assistant$ python3 -m script.hassfest
Traceback (most recent call last):
  File "/usr/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/mnt/c/dd/repos/home-assistant/script/hassfest/__main__.py", line 5, in <module>
    from . import (
  File "/mnt/c/dd/repos/home-assistant/script/hassfest/codeowners.py", line 4, in <module>
    from .model import Config, Integration
  File "/mnt/c/dd/repos/home-assistant/script/hassfest/model.py", line 7, in <module>
    import attr
ModuleNotFoundError: No module named 'attr'
(venv) oscar@Oscar-Surface:/mnt/c/dd/Repos/home-assistant$ python3 -m script.hassfest
```
What am I missing?

zxdavb committed:
switch evohome to use a whitelist for valid zonetype (#31047)

ocalvo committed:
Update code owners

springstan commented:
Have you installed all requirements via `script/setup`?

ocalvo committed:
Update requiriments

ocalvo commented:
> Have you installed all requirements via `script/setup`?

I figure it out (I am an idiot), my other terminal did not have the environment active.

frenck committed:
Bump pytest to 5.3.4 (#31045)

ocalvo commented:
PyLint is failling with:
```
ing python-gammu==2.12
  Downloading python-gammu-2.12.tar.gz (135 kB)
    ERROR: Command errored out with exit status 100:
     command: /__w/1/s/venv/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-9mnlhhrs/python-gammu/setup.py'"'"'; __file__='"'"'/tmp/pip-install-9mnlhhrs/python-gammu/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-install-9mnlhhrs/python-gammu/pip-egg-info
         cwd: /tmp/pip-install-9mnlhhrs/python-gammu/
    Complete output (9 lines):
    Package gammu was not found in the pkg-config search path.
    Perhaps you should add the directory containing `gammu.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'gammu' found
    Package gammu-smsd was not found in the pkg-config search path.
    Perhaps you should add the directory containing `gammu-smsd.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'gammu-smsd' found
    Can not find supported Gammu version using pkg-config!
    ----------------------------------------
ERROR: Command errored out with exit status 100: python setup.py egg_info Check the logs for full command output.
##[error]Bash exited with code '1'.
```

frenck committed:
Iteration on issue and PR templates (#31058)

* Iteration on issue and PR templates

* Some small improvements to make it look better in the PR editor

* Better English

* Please keep template

* Consistency

* Adds www to website link, to make more clear it is our website

* Process review suggestion by Martin

* Added dep bump as type of change

* English

ocalvo committed:
Update azure pipeline

ocalvo commented:
@pvizeli PyLint is failing to install becuase gammu-dev is missing. 
I need help updating the [azp-step-cache.yaml pipeline](https://github.com/home-assistant/ci-azure/blob/810c943805418b71b4f80655b32332161c07c675/templates/azp-step-cache.yaml)

ocalvo commented:
> Like I said, if this is merged to code base, I will care about the wheels
@pvizeli It seems that I wont be able to pass PyLint without your help.

joegross committed:
Fix prometheus component to fully sanitize Unicode characters (#31037)

* Fix prometheus component to fully santize Unicode characters.

Sanitization used isdigit() to match numerics but that also matches 
Unicode "digit" characters that require special handling, like 
superscripts. Failures would look like this:

ValueError: Invalid metric name: sensor_unit_u0x23_per_m³

Changed sanitize to use string.digits, which matches only ascii digits, 
and added test.

See 
https://stackoverflow.com/questions/44891070/whats-the-difference-between-str-isdigit-isnumeric-and-isdecimal-in-python 
for discussion.

https://docs.python.org/3/library/stdtypes.html#str.isdigit

https://docs.python.org/3.7/library/string.html#string.digits

* fix formatting to comply with black

jensihnow committed:
Add tahoma io:AwningValanceIOComponent (#30944)

springstan committed:
Fix fan_modes in tuya climate (#30942)

homeassistant committed:
[ci skip] Translation update

reuank committed:
Fix luftdaten broken icon for "Pressure at sealevel" (#31053)

Quentame committed:
Do not stop fetching iCloud data after service error (#31069)

raman325 committed:
Enhance info_from_service function in zeroconf integration (#31059)

* enhance zeroconf service info decoding to include raw bytes

* Update homeassistant/components/zeroconf/__init__.py

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

* fix test based on last commit

* fix test based on last commit

* remove .keys() when asserting processed and raw service info properties

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

raman325 committed:
Add zeroconf discovery support for vizio integration (#30949)

* add missing tests

* readd removed test

* add zeroconf discovery support for vizio integration

* no mock_coro_func needed

* add reasonable timeout and don't log exceptions from pyvizio due to timeout

* add test to test options update and bump pyvizio to avoid timeout issues

* update requirements_*

* fix gaps in coverage

* change return hint for async_setup_entry

* use source variables instead of strings

* only get unique ID if about to create entry

* update based on review

* Revert "update based on review"

This reverts commit 0d612a90eb7d02c92061f902973e527267e3110a.

* f-string

* fix last review

* revert cleanup changes to simplify PR

* remove unnecessary ConfigFlow object variables to simplify logic

* revert cleanup changes to make review easier, noted for future cleanup

* revert cleanup changes to make review easier, noted for future cleanup

* move zeroconf service type constant to test module

d0ugal committed:
Add the Home Assistant version as a Sentry release (#31065)

This helps make it easier to identify which version of Home Assistant
first introduces a error.

frenck committed:
Add disabled entities support to WLED (#31040)

pvizeli commented:
@ocalvo that's fine. If you fix all review commands and it's a step close to merge it, I update the azure CI to support it. You can run pylint local and fix the issues, so it's not blocking you later.

dmulcahey committed:
change group id creation (#31075)

ocalvo commented:
> @ocalvo that's fine. If you fix all review commands and it's a step close to merge it, I update the azure CI to support it. You can run pylint local and fix the issues, so it's not blocking you later.

Thanks, how do i fix this error locally:
```
************* Module homeassistant.components.sms
homeassistant/components/sms/__init__.py:4:0: E0401: Unable to import 'gammu' (import-error)
ERROR: InvocationError for command /mnt/c/dd/repos/home-assistant/.tox/pylint/bin/pylint homeassistant (exited with code 2)
```
?

balloob committed:
Allow unloading mobile app (#30995)

bendavid committed:
Improve state tracking for WebOsTV (#31042)

* upgrade to aiopylgtv 0.3.0 and corresponding simplification and cleanup of webostv state tracking

* properly handle case where Live TV is not reported in list of apps

* fix tests (entity state is no longer linked to source id)

* fix pylint checks

* avoid unnecessary retrieval of channel list

* use only standard home assistant states

thoscut committed:
Add DIRECTORY and PLUGIN to kodi media types (#28336)

ochlocracy committed:
Update AdjustRange Handler Service Calls. (#31016)

Add a AlexaGlobalCatalog value to all labels.

bieniu committed:
Add zeroconf discovery support to Brother Printer integration (#30959)

* Add zeroconf discovery support

* Fix data for config_entry

* Add sting for zeroconf confirm dialog

* Add and fix tests

* Fix pylint errors

* Suggested changes

* Tests

* Remove unnecessary object

* Add error handling

* Remove unnecessary objects

* Suggested change

* Suggested change

* Use core interfaces for tests

pvizeli commented:
Please read carefully https://developers.home-assistant.io/docs/en/next/creating_component_code_review.html#3-componentplatform-communication and change the code again this. After that, I give a shot for review.

ocalvo commented:
> Please read carefully https://developers.home-assistant.io/docs/en/next/creating_component_code_review.html#3-componentplatform-communication and change the code again this. After that, I give a shot for review.

Just to confirm, you mean that in order to share the gammu state machine, I should use hass.data["SMS"]?
This make sense to me.

balloob committed:
Migrate zone to use collection helpers (#30774)

homeassistant committed:
[ci skip] Translation update

raman325 committed:
Code Cleanup for Vizio component (#31076)

* code cleanup

* dont use named arguments for positional arguments

* remove extra comma

ocalvo committed:
Apply PR feedback

balloob committed:
Simplify automation services (#30996)

* Simplify automation services

* Empty commit to re-trigger build

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

bachya committed:
Remove monitored conditions from OpenUV (#31019)

* Remove monitored conditions from OpenUV

* Code review comments

jkeljo committed:
Add myself to owners for components I contributed (#31020)

* Add myself to owners for components I contributed

* Update CODEOWNERS

springstan committed:
Deprecate hook integration (#31046)

jameshilliard committed:
Add keep-alive which is required for some hlk-sw16 variants (#31062)

fredrike committed:
pydaikin version bump (#31080)

ocalvo committed:
Add missing test requirement

ocalvo committed:
Undo prev change

bachya committed:
Remove monitored conditions from RainMachine (#31066)

* Remove monitored conditions from RainMachine

* Migrate config entry

* Revert "Migrate config entry"

This reverts commit 84fcf5120ff317d088761aff70402608d58d7175.

* Code review comments

* Disable some entities by default

ocalvo committed:
Update lines

robmarkcole committed:
Add sighthound integration (#28824)

* Add component files

* Add test state

* Adds person detected event

* Update CODEOWNERS

* Updates requirements

* remove unused datetime

* Bump sighthound version

* Update CODEOWNERS

* Update CODEOWNERS

* Create requirements_test_all.txt

* Address reviewer comments

* Add test for bad_api_key

frenck committed:
Cleanup of PR and issue templates (#31070)

* Remove old issue template

* Small wording improvement

* Consistency

* Configuration fix

* Process review suggestions

pvizeli commented on .coveragerc:
```diff
@@ -640,6 +640,8 @@ omit =
     homeassistant/components/smarty/*
     homeassistant/components/smarthab/*
+    homeassistant/components/sms/__init__.py
+    homeassistant/components/sms/notify.py
```
`../*`

pvizeli commented on homeassistant/components/sms/__init__.py:
```diff
@@ -0,0 +1,22 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+
```
Missing config schema

pvizeli commented on homeassistant/components/sms/__init__.py:
```diff
@@ -0,0 +1,22 @@
+    state_machine.SetConfig(0, dict(Device=device, Connection="at"))
+    state_machine.Init()
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][STATE_MACHINE] = state_machine
```
`hass.data[DOMAIN] = state_machine`

pvizeli reviewed (changes requested):
Please don't call this `state_machine` because it's confused with our state machine. Call it `modem` or `sms_gateway` or something else.

frenck committed:
Upgrade sqlalchemy to 1.3.13 (#31101)

mindigmarton committed:
Update emulated_roku to 0.2.1 (#31100)

dokmic committed:
Fix HomeKit window covering to support float numbers in the position (#31081)

* Fix HomeKit window covering to support float numbers in the position

* Fix HomeKit window covering to cast current position to an integer value

tetienne committed:
Add temperature support to light template (#30595)

* Add temperature support

* Use guard clause

ocalvo committed:
Apply PR feedback

uvjustin committed:
Allow framerates less than 1 (#31108)

ocalvo committed:
Update for pylint

ocalvo committed:
Update for pylint

fabaff committed:
Upgrade shodan to 1.21.3 (#31111)

frenck committed:
Add update_entry dict to unique ID flow abort helper (#31090)

* Add update_entry dict to unique ID flow abort helper

* Process review suggestions

* Process review suggestions

* Add additional test

tulindo committed:
Change Samsung TV state detection (#30236)

* Changed Samsung TV state detection

* Trying codecov fix

* Update Rewritten

* Changed update method

* Timeout handling

* Fixed autodetect tests

* Fixed last test

* Added test to complete codecov

* Removed state settings in send_key method

* Fixed pylint

* Fixed some tests

* codecov fix

Poeschl committed:
Use speak2mary for MaryTTS integration and enable sound effects (#30805)

* Use speak2mary for MaryTTS integration and enable sound effects

* Replace static defaults for effects with user configured ones

frenck committed:
Add disabled entities support to AdGuard (#31106)

homeassistant committed:
[ci skip] Translation update

raman325 committed:
Fix connection failure log message in async_setup_entry for Vizio component (#31097)

* remove connection check during setup since it is already done during config flow

* revert change and fix log message

* demote connection setup to warning

BKPepe committed:
Upgrade youtube_dl to version 2020.01.24 (#31124)

balloob committed:
Update cast to 4.1.1 (#31113)

pvizeli commented on homeassistant/components/sms/__init__.py:
```diff
@@ -2,21 +2,27 @@
 _LOGGER = logging.getLogger(__name__)
 
+CONFIG_SCHEMA = vol.Schema(
+    {DOMAIN: vol.Schema({vol.Required(CONF_DEVICE): cv.string})}, extra=vol.ALLOW_EXTRA,
```
`cv.isdevice` instead `cv.string`

pvizeli commented on homeassistant/components/sms/notify.py:
```diff
@@ -0,0 +1,43 @@
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {vol.Required(CONF_PHONE_NUMBER): cv.string, vol.Optional(CONF_NAME): cv.string}
```
All our exists SMS integration use `recipient`. Please adapt this and rename `phone_number`

pvizeli commented on homeassistant/components/sms/__init__.py:
```diff
@@ -0,0 +1,28 @@
+    gateway.SetConfig(0, dict(Device=device, Connection="at"))
+    gateway.Init()
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN] = gateway
```
Just `hass.data[...] = gateway`

pvizeli commented on homeassistant/components/sms/__init__.py:
```diff
@@ -0,0 +1,28 @@
+    device = conf.get(CONF_DEVICE)
+    gateway = gammu.StateMachine()  # pylint: disable=no-member
+    gateway.SetConfig(0, dict(Device=device, Connection="at"))
+    gateway.Init()
```
Any error handling around this to catch issues with device?

pvizeli reviewed (changes requested):
Update https://github.com/home-assistant/home-assistant/blob/dev/script/gen_requirements_all.py#L15 with your library name.

Update azure-pipelines-wheels.yml to uncomment that line again.

lade-odoo committed:
Add Buienradar camera for Belgium (#30399)

* Buienradar Camera for Belgium

* Voluptuous check for country code

* Black formatting

* Testing for Buienradar Belgium

* Changed MULTIPLE CHOICE variable name

* Changes from frenck review

dokmic committed:
Fix Template components to process entity_id configuration option (#31084)

* Fix Template Cover component to process entity_id configuration option

* Fix Template Light component to process entity_id configuration option

* Empty commit to re-trigger build

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

ocalvo committed:
Apply PR feedback

Commit:
Remove unused async_setup_platform (#31132)

ocalvo committed:
Apply PR feedback

frenck committed:
Update Hue existing config entry with discovery data (#31087)

* Update Hue existing config entry with discovery data

* Updated method documentation comments

* Update implementation to match latest dev

* Use named argument for clarity

ocalvo committed:
Apply PR feedback

frenck committed:
Rewrite of Spotify integration (#30717)

* Rewrite of Spotify integration

* Update homeassistant/components/spotify/config_flow.py

Co-Authored-By: Paulus Schoutsen <balloob@gmail.com>

* Remove configurator dependency

* Strip whitespace from device model in case Spotify product is missing

* Ensure domain dict exists in hass data on setup entry

* Simply config validation for client id and secret

* Abort flow on any exception from spotipy

* Add tests for config flow

* Gen requirements all

* Add test package __init__

* Remove Spotify from coveragerc

* Made alias handling more robuust

* Fix supported_features for Spotify free and open accounts

* Improve error message in the logs

* Re-implement Spotify media_player

* Change media content type when play a playlist

* Process review suggestions

* Move Spotify init, static current user and supported_features

* Remove unneeded me call

* Remove playlist content type due to frontend issues

* Improve playlist handling, when context is missing

* Handle entity disabled correctly

* Handle being offline/unavailable correctly

* Bump Spotipy to 2.7.1

* Update coverage RC, mark integration silver

* Remove URI limitation, lib supports all Spotify URI's now

* Final cleanup

* Addresses Pylint error

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

ocalvo committed:
Fix lynt

balloob committed:
Validate coveragerc with hassfest (#31112)

* Validate coveragerc

* Test if files exists

* Print progress

* Flush

ocalvo committed:
Update manifest.json

scop committed:
Make pylint fail on informational messages too (#31136)

Refs https://github.com/PyCQA/pylint/issues/3250

ocalvo committed:
Update azure-pipelines-wheels.yml

frenck committed:
Add logo & icon support to Manifest (#31131)

* Add logo & icon support to Manifest

* Add URL validation

balloob committed:
Fix when you have two wemo devices (#31139)

homeassistant committed:
[ci skip] Translation update

ochlocracy committed:
Include supported states in Alexa SecurityPanelController configuration object (#31120)

* Update Security Panel Controller.

* Update Security Panel Controller.

* Sort imports.

bachya committed:
Make SimpliSafe integration more resilient to SimpliSafe cloud issues (#31116)

* Make SimpliSafe integration more resilient to SimpliSafe cloud issues

* Clear emergency refresh token

* Stop listening when appropriate

* Cleanup

* Saving refresh token should happen after all updates

* Code review

bachya committed:
Add smarter API usage for RainMachine (#31115)

* Make RainMachine smarter with API usage

* Remove debug statements

* Fix deregistration

* Code review comments

* Code review

* Use an asyncio.Lock

* Remove unnecessary guard clause

* Ensure registation lock per API category

bachya committed:
Fix RainMachine update action (#31147)

scop committed:
Fix Huawei LTE SMS recipient setting from options UI (#31117)

* Fix Huawei LTE SMS recipient setting from options UI

Refs https://github.com/home-assistant/home-assistant/issues/30827

* Use core interfaces in tests

* ...more

scop committed:
Remove no longer used Hound config (#31154)

Adminiuga committed:
Update ZHA remotes registry to proper identify "remotes (#31146)

* Update ZHA remote device types.

* Remove `binary_sensor` entity from affected devices.

* Update ZHA Profiles.

Prevent DeviceType.ON_OFF_LIGHT_SWITCH from creating entities.

* Update tests and remove unused entities.

billyburly committed:
improvements to zha cover (#31144)

bendavid committed:
Fix state handling for older webos versions (#31099)

* fix state handling for older webos versions

* update aiopylgtv to 0.3.2

Quentame committed:
Separate iCloud class (#31022)

* Separate iCloud class

* Update .coveragerc

* Fix pipe

oblogic7 committed:
Remove Owlet component (#31160)

* Remove owlet component

* Remove owlet from requirements_all.txt

dingusdk committed:
Fix the ihc.set_runtime_value_int service function not working with templates (#31145)

* Make the set_runtime_value_int function work with template values

* Use newest version of the ihcsdk library

* Make the set_runtime_value_int function work with template values

* Use newest version of the ihcsdk library

* Updated to the newest ihcsdk 2.5.0

* Formatted changes to make it pass CI tests

homeassistant committed:
[ci skip] Translation update

bachya committed:
Add additional alarm states to SimpliSafe (#31060)

* Add additional alarm states to SimpliSafe

* Remove unused constant

* Remove redundant local variable

bachya committed:
Refactor RainMachine switch platform (#31148)

* Import constants sanely

* Linting

* Rename data constants for consistency

* Refactor RainMachine switch platform

* Comments

* Cleanup

* Refactor switch and sensor API calls to be separate

* Linting

* Make sure zones are updated in appropriate service calls

* Correctly decrement

* Linting

* Don't do weird inheritance

* Ensure service calls update data properly

* Docstring

* Docstring

* Errors can be logged without string conversion

* Code review comments

JeffLIrion committed:
Implement 'volume_set' service for Android TV devices (#31161)

inverse committed:
Introduce unique_id for lastfm to allow changing entity_id in backwards compatible way (#31163)

* Replace . with _ for lastfm entity_id

* lint

* double quotes

* Rollback change, add unique_id

* Expose prop

* Generate unique ID from user

* Linty

* FIx linter

* Revert changes for splitting entity_id

* Simplify

cereal2nd committed:
Fix Velbus covers (includes velbus lib upgrade) (#31153)

* Fix velbus covers

* Update python-velbus lib

* flake8 and black fixes

* Fix comments

* fix codeowner

Commit:
Use  unique_id for config_entries of HomematicIP Cloud (#31133)

* use hapid as unique_id for config_entry of HomematicIP Cloud

* Add qualita_scale to manifest

* Update config_flow

* use core interface for config_flow tests

* refactor to fail earlier

* use asynctest

* rewrite tests

* rewrite tests

* fix test

* add assert

brefra committed:
Add counters sensors for Velbus (#31165)

* Add counter sensor

* Apply black formatting

* Move all counter logic to sensor.py

* Code cleanup

teharris1 committed:
Bump insteonplm to 0.16.6 (#31182)

ocalvo committed:
Update manifest.json

frenck committed:
Add hassfest URL validation to documentation link (#31143)

ocalvo committed:
Update azure-pipelines-wheels.yml

finnysamuel committed:
Bump aiobotocore to 0.11.1 (#30951)

* Bumped aiobotocore version to 0.11.1
* New dependencies are added to requirements_all.txt

bachya committed:
Use non-deprecated method of instantiating RainMachine client (#31149)

scop committed:
Upgrade huawei-lte-api to 1.4.7 (#31155)

https://github.com/Salamek/huawei-lte-api/releases/tag/1.4.7

balloob committed:
Add unique IDs to automation/scenes (#31150)

* Add unique IDs to automation and scenes

* Fix typo

starkillerOG committed:
Add webostv sound_output capability (#31121)

* Webostv: add sound_output capability

Add the ability to read and set the sound_output

* Webostv: add sound_output capability

* Webostv: add sound_output capability

* fix blank spaces

* fix to long line

* add ,

* Import ATTR_SOUND_OUTPUT

Do not have the ability to test this change right now

* Add white space

* Create const.py

* Use const import

* Use import from const.py

* Add docstring

* Change order

* Change order

* Fix import

* Fix import

* Fix typo

* Change order again

* Change order again

* Change order of attributes

rohankapoorcom committed:
Check that documentation urls are valid (#31188)

* Check that documentation urls are valid

* Validate documentation url in pieces

frenck committed:
Disable failing dsmr tests (#31202)

* Disable failing dsmr tests

* Disable module, disable import of missing dep

alandtse committed:
Fix unnecessary regeneration of access token in Tesla component (#31193)

* Fix unnecessary regeneration of access token

* Add manifest.json

frenck committed:
Upgrade HAP-python to 2.7.0 (#31201)

cyberjunky committed:
Added missing file (#31189)

dmulcahey committed:
update remove service (#31164)

ktnrg45 committed:
Fix ps4 errors if pin begins with a 0 (#31198)

* Fix errors if pin begins with a 0

* Test PIN leading with zero

* Edit tests

Quentame committed:
Use config_entry.unique_id in Linky (#31051)

* Use config_entry.unique_id in Linky

* Reviews

* _show_setup_form not async

cyberjunky committed:
Add Garmin Connect integration (#30792)

* Added code files

* Correctly name init file

* Update codeowners

* Update requirements

* Added code files

* Correctly name init file

* Update codeowners

* Update requirements

* Black changes, added to coveragerc

* Removed documentation location for now

* Added documentation url

* Fixed merge

* Fixed flake8 syntax

* Fixed isort

* Removed false check and double throttle, applied time format change

* Renamed email to username, used dict, deleted unused type, changed attr name

* Async and ConfigFlow code

* Fixes

* Added device_class and misc fixes

* isort and pylint fixes

* Removed from test requirements

* Fixed isort checkblack

* Removed host field

* Fixed coveragerc

* Start working test file

* Added more config_flow tests

* Enable only most used sensors by default

* Added more default enabled sensors, fixed tests

* Fixed isort

* Test config_flow  improvements

* Remove unused import

* Removed redundant patch calls

* Fixed mock return value

* Updated to garmin_connect 0.1.8 fixed exceptions

* Quick fix test patch to see if rest is error free

* Fixed mock routine

* Code improvements from PR feedback

* Fix entity indentifier

* Reverted device identifier

* Fixed abort message

* Test fix

* Fixed unique_id MockConfigEntry

bachya committed:
Constrain SimpliSafe's check for emergency token usage (#31214)

balloob committed:
Update Hue discovery (#31215)

cclauss committed:
Fix typo: serivce --> service (#31217)

Commit:
Cleanup of HomematicIP Cloud tests (#31181)

* CleanUp tests for HomematicIP_Cloud

* Remove not required CoroutineMock

* remove None return in mocks, add asserts

* rewrite test

aaska committed:
Bump python-synology to 0.4.0 : Add support for DSM v5 + fix sensors unknown for 15 min (#31049)

* updating new api version

* Added new configuration option for updated API

pvizeli commented:
Please address pylint issues

pvizeli commented on homeassistant/components/sms/__init__.py:
```diff
@@ -0,0 +1,32 @@
+        gateway.Init()
+        hass.data[DOMAIN] = gateway
+        return True
+    except gammu.GSMError as exc:  # pylint: disable=no-member
```
Please only around that code which the error can trow:
```
try:
   gateway.SetConfig(...)
   gateway.Init()
except ...
```

ocalvo replied:
I am working on it. Please give me more time.

ocalvo commented:
> Please address pylint issues

According to tox and azure pipelines (before the change to azure-wheels.yaml) the code was pylint clean.
How do I look at the specific errors?

pvizeli commented:
https://github.com/home-assistant/home-assistant/pull/31035/checks?check_run_id=409808520

pvizeli commented:
If you address my last comment and fix the lint issues, we can merge before beta cut, otherwise it's comes with 0.106

ocalvo commented:
> https://github.com/home-assistant/home-assistant/pull/31035/checks?check_run_id=409808520

But this is a failure unrelated to my change:
```
ImportError while importing test module '/__w/1/s/tests/components/dsmr/test_sensor.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/components/dsmr/test_sensor.py:17: in <module>
    from homeassistant.components.dsmr.sensor import DerivativeDSMREntity
homeassistant/components/dsmr/sensor.py:8: in <module>
    from dsmr_parser.clients.protocol import create_dsmr_reader, create_tcp_dsmr_reader
venv/lib/python3.7/site-packages/dsmr_parser/clients/__init__.py:3: in <module>
    from dsmr_parser.clients.serial_ import SerialReader, AsyncSerialReader
venv/lib/python3.7/site-packages/dsmr_parser/clients/serial_.py:8: in <module>
    from dsmr_parser.parsers import TelegramParser
venv/lib/python3.7/site-packages/dsmr_parser/parsers.py:4: in <module>
    from PyCRC.CRC16 import CRC16
E   ModuleNotFoundError: No module named 'PyCRC'
```

ocalvo commented:
> If you address my last comment and fix the lint issues, we can merge before beta cut, otherwise it's comes with 0.106

I will submit in the next 15 minutes.

pvizeli commented:
you need rebase it to latest dev

ocalvo committed:
Merge pull request #2 from home-assistant/dev

Update from dev

ocalvo committed:
Apply PR feedback

homeassistant commented:
Hello @ocalvo,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`ad6f75d6be2508d74404e59d87a752da00afcfa0`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;31035&#x2F;commits&#x2F;ad6f75d6be2508d74404e59d87a752da00afcfa0) - This commit has something that looks like an email address (root@ocalvo-server02.ntdev.corp.microsoft.com). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

ocalvo committed:
Apply PR feedback

ocalvo committed:
Merge branch 'sms' of https://github.com/ocalvo/home-assistant into sms

homeassistant commented:
Hello @ocalvo,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`5fa04f0df6b04d82e9e6d1fc0c9410ae19c50e06`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;31035&#x2F;commits&#x2F;5fa04f0df6b04d82e9e6d1fc0c9410ae19c50e06) - This commit has something that looks like an email address (root@ocalvo-server02.ntdev.corp.microsoft.com). Maybe try linking that to GitHub?.
- [`ad6f75d6be2508d74404e59d87a752da00afcfa0`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;31035&#x2F;commits&#x2F;ad6f75d6be2508d74404e59d87a752da00afcfa0) - This commit has something that looks like an email address (root@ocalvo-server02.ntdev.corp.microsoft.com). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

ocalvo commented:
Need to restart this PR.
